### PR TITLE
systemd: fix global scope

### PR DESCRIPTION
**Pull request by tchernomax**
# SUMMARY

major rewrite of systemd.py

I start with #55936 but discovered that:

1. I was wrong
2. other parts of the code was also wrong

# ISSUE TYPE

    Bugfix Pull Request

# COMPONENT NAME

systemd

### CODE REVIEW

**Comment by tchernomax** [comment]
ping @bcoca

File: lib/ansible/modules/system/systemd.py [hunk 1]
```diff
@@ -395,134 +387,166 @@ def main():
...
+        # we also considere it a failure if it returns nothing (I don't know if it's possible)
+        rc = 1
+
+    if rc == 0:
+        # load return of systemctl show into dictionary for easy access and return
+        result['status'] = parse_systemctl_show(to_native(out).split('\n'))
+
+        is_found = result['status'].get('LoadState', 'not-found') != 'not-found'
+        is_masked = result['status'].get('LoadState', 'not-found') == 'masked'
+        # UnitFileState is not available on old systemd version (pre 230,
```
Comment (jamescassell) [inline lib/ansible/modules/system/systemd.py:415]: Which version is in RHEL 7?

**Review by jamescassell — APPROVED** [review]
shipit

Would be good to have a better explanation for each change.

File: lib/ansible/modules/system/systemd.py (hunk 1 above)
Comment (tchernomax) [reply lib/ansible/modules/system/systemd.py:415]: @jamescassell It seam RHEL 7 use systemd 219.

(thanks for the review)

**Comment by jborean93** [comment]
We are unsure what this particular PR fixes with the systemd module. Due to the age of it, the conflicts with the changes we are going to close this. If this is still a problem we can review them in a new PR with changes rebased against the code in devel.

**Pull request closed without merging**